### ISSUE TITLE
[Question] Base AMI bootstraping

### ISSUE THREAD

**Issue by btd** [segmentio/stack#6]
[In your bootstrap scripts](https://github.com/segmentio/stack/blob/master/packer/base/root/usr/local/bin/bootstrap) i noticed you directly set all values. Have you tried to use cloud-init for this? (I am asking just because we use ubuntu 14.04 and cloud-init for the same). Maybe you already tried and it does not work with 16.04/systemd?

Thanks.

**Comment by achille-roussel** [comment]
We did try, it mostly worked but we ran into race conditions sometimes because we couldn't enforce the execution ordering of cloud-init vs systemd managed processes.
It may work just fine depending on the use case, we had some internal services that picked the hostname when they started and therefore had an invalid value.

**Comment by btd** [comment]
I totally agree, to fix race condition with upstart i used `start on started rc` (because at 14.04 they use still sysv init scripts for cloud-init which are executed with another upstart job).

**Comment by gjohnson** [comment]
@achille-roussel did you try depending on `cloud-final.service` in the [bootstrap unit](https://github.com/segmentio/stack/blob/0dfeb007ee052d54a2cf75864700afc65482b441/packer/base/root/etc/systemd/system/bootstrap.service)?

**Comment by achille-roussel** [comment]
I did not, although I feel like that could make the boot time much longer, right now cloud-init takes quite a while to complete.
